theme/scripts/patronSlider.ts

```diff
@@ -0,0 +1,161 @@
+              // do not render a pip for that value.
+              if ([5, 10, 15, 20, 25].indexOf(value) > -1) {
+                return 1;
+              }
+
+              // Render a pip.
+              return -1;
+            },
+          },
+        }
+      );
+
+      const handle = document.querySelector(PatronSlider.SELECTORS.handle);
+      const patronTierLink = document.querySelector(
+        PatronSlider.SELECTORS.patronTierLink
+      );
+      const tierCopywriting = document.querySelector(
+        PatronSlider.SELECTORS.patronTierDescription
+      );
+      const patronTierName = document.querySelector(
+        PatronSlider.SELECTORS.patronTierName
+      );
+
+      // Used to set the content of our slider handle
+      handle.setAttribute("data-before", `$${PatronSlider.START_VALUE}`);
+      // Used to set the content of our drag handle.
+      handle.setAttribute("data-after", PatronSlider.MONTHLY);
+
+      patronTierLink.setAttribute(
+        "href",
+        PatronSlider.PATRON_TIERS[PatronSlider.START_VALUE].href
+      );
+
+      tierCopywriting.textContent =
+        PatronSlider.PATRON_TIERS[PatronSlider.START_VALUE].description;
+
+      patronTierName.textContent =
+        PatronSlider.PATRON_TIERS[PatronSlider.START_VALUE].name;
+
+      PatronSlider.bindEventListeners();
+    },
+
+    bindEventListeners() {
+      const pips = document.querySelectorAll(PatronSlider.SELECTORS.pips);
+      const handle = document.querySelector(PatronSlider.SELECTORS.handle);
+      const patronTierLink = document.querySelector(
+        PatronSlider.SELECTORS.patronTierLink
+      );
+      const tierCopywriting = document.querySelector(
+        PatronSlider.SELECTORS.patronTierDescription
+      );
+      const patronTierName = document.querySelector(
+        PatronSlider.SELECTORS.patronTierName
+      );
+
+      // Handle clicks on "pips", or notches, on the slider.
+      const handlePipClick = (e) => {
+        const valueEl = getNextSibling(
+          e.target,
+          PatronSlider.SELECTORS.pipsValue
+        );
+        var value = Number(valueEl.getAttribute("data-value"));
+        PatronSlider.SLIDER.set(value);
+      };
+
+      // When the slider hits a notch, even before the mouse click is released.
+      PatronSlider.SLIDER.on("update", function (event) {
+        const sliderValue = parseInt(event.toString());
+
+        handle.setAttribute("data-before", `$${sliderValue}`);
+
+        tierCopywriting.textContent =
+          PatronSlider.PATRON_TIERS[sliderValue].description;
+
+        patronTierName.textContent =
+          PatronSlider.PATRON_TIERS[sliderValue].name;
+
+        patronTierLink.setAttribute(
+          "href",
+          PatronSlider.PATRON_TIERS[sliderValue].href
+        );
+      });
+
+      // Listen for clicks on our pips
+      for (var i = 0; i < pips.length; i++) {
+        pips[i].addEventListener("click", (e) => handlePipClick(e));
```

Comment:
You should be able to just pass `handlePipClick` as the callback, instead of having to create an anonymous function each time

Reply:
I always seem to forget that you can do this with callback functions. It's still so cool every time. Is there a term for this so I can learn more? I googled something like `implicit callback parameters in javascript` but could not find what I was looking for.

Reply:
`handlePipClick` is an "uninvoked" function
`(e) => handlePipClick(e)` is an "uninvoked" function (ie, it is not being called, just declared at this point)

Both are a type of variable. When the event listener fires, it will call the callback with it's `e` argument. Because both of the variants above, take an `e` argument, they are compatible, so you don't need to wrap the original